packages/nc-gui/components/cell/DatePicker.vue

```diff
@@ -76,14 +77,89 @@ useSelectedCellKeyupListener(active, (e: KeyboardEvent) => {
   switch (e.key) {
     case 'Enter':
       e.stopPropagation()
-      open.value = true
+      // skip if drawer / modal is active
+      if (isDrawerOrModalExist()) {
+        return
+      }
+      if (!open.value) {
+        // open date picker
+        open.value = true
+      } else {
+        // select the current day
+        const el = document.querySelector('.nc-picker-date.active .ant-picker-cell-selected') as HTMLButtonElement
+        if (el) {
+          el.click()
+          open.value = false
+        }
+      }
       break
     case 'Escape':
+      // skip if drawer / modal is active
+      if (isDrawerOrModalExist()) {
+        return
+      }
       if (open.value) {
         e.stopPropagation()
         open.value = false
       }
       break
+    case 'ArrowLeft':
+      if (!localState) {
+        ;(document.querySelector('.nc-picker-date.active .ant-picker-header-prev-btn') as HTMLButtonElement)?.click()
```

Comment:
`;` in the start? Linting bugging out?

Reply:
Nope. This is actually required because of the round brackets. Think about the following case

```js
const foo = 'bar'
(document.querySelector('.nc-picker-date.active .ant-picker-header-prev-btn') as HTMLButtonElement)?.click()
```

without `;`, it would be

```js
const foo = 'bar'(document.querySelector('.nc-picker-date.active .ant-picker-header-prev-btn') as HTMLButtonElement)?.click()
```

Now you know why `;` is there.

Reply:
Got it 👍